### PR TITLE
CVSL-1906 add endpoint for get approval caseload

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to add an endpoint to call the new approver service to get the approval licences.

### CODE REVIEW

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/CaseloadControllerTest.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    )
+
+    val approvalCase = ApprovalCase(
+      licenceId = 1L,
+    )
+
+    whenever(approverCaseloadService.getApprovalNeeded(request)).thenReturn(listOf(approvalCase))
+
+    val response = mvc.perform(
+      post("/caseload/get-approval-needed")
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/CaseloadControllerTest.kt:75]: **suggestion:** could group by user `/caseload/prison-approver/approval-needed` ?

Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/CaseloadControllerTest.kt:75]: or: `/caseload/prison-approvals/to-review ?`

Comment (gurps-singh) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/CaseloadControllerTest.kt:75]: I prefer the first suggestion as it is clear who the caseload is for and what the caseload is for.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**